src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -307,19 +308,30 @@ def run_query(
         to_pandas: bool = False,
         to_iter: bool = False,
         is_ddl_on_temp_object: bool = False,
+        block: bool = True,
+        data_type: AsyncDataType = AsyncDataType.ROW,
```

Comment:
Better put them into kwargs

Reply:
I think we should keep it this way because kwargs will be passed into cursor.execute() in the end while these args are not needed in it.